core/reactor_common.c

```diff
@@ -600,7 +641,11 @@ lf_token_t* _lf_initialize_token(lf_token_t* token, size_t length) {
     void* value = malloc(token->element_size * length);
     // Count allocations to issue a warning if this is never freed.
     _lf_count_payload_allocations++;
-    return _lf_initialize_token_with_value(token, value, length);
+    lf_token_t* result = _lf_initialize_token_with_value(token, value, length);
+    // Make sure the value can be freed.
+    if (result->ok_to_free == neither_token_nor_value) result->ok_to_free = value_only;
+    if (result->ok_to_free == token_only) result->ok_to_free = token_and_value;
```

Comment:
here again, we could simply mask with value flag and value freeing would be enforced.